core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -1105,32 +1084,33 @@ object GroupMetadataManager {
    * Generates the payload for offset commit message from given offset and metadata
    *
    * @param offsetAndMetadata consumer's current offset and metadata
+   * @param apiVersion the api version
    * @return payload for offset commit message
    */
-  private[group] def offsetCommitValue(offsetAndMetadata: OffsetAndMetadata): Array[Byte] = {
+  private[group] def offsetCommitValue(offsetAndMetadata: OffsetAndMetadata,
+                                       apiVersion: ApiVersion): Array[Byte] = {
     // generate commit value according to schema version
+    val (version, value) = {
+      if (apiVersion < KAFKA_2_0_IV2)
```

Comment:
I think I'm missing something again. If we are dealing with an old OffsetCommit which contains a custom retention time, don't we need to use v1 also. Do we have a test case for this scenario?